This project is aimed at generating a multilocus phylogeny for North American species in sect. Amanita (subgen Amanita, genus Amanita Although this is a phylogenetics project, it's ultimate goal is to interrogate the taxonomy of this section, to see how applied names compare with apparent phylogenetic signal.

**Data**: 
* publically available GenBank accessions from several species in section _Amanita_. See `./RawData/GB_acc_no` for the accession numbers and taxa sampled. For each taxon, I have tried to get data across all four loci (ITS1, LSU, $\beta$-Tubulin, and EF1-$\alpha$)
* Four loci data extracted from 25 **unpublished** *Amanita muscaria* genomes from across the globe, including invasive populations
* Four loci data from _A. rubescens_ (subgen. _Amanitina_ sect. _Validae_ ), which I have chosen for it's static relationship to the members of this section.  
* GenBank Accessions were currated manually and downloaded into four `.fasta.` files. These do not have quality scores and were likely generated by sanger sequencing, making automated QC impossible. Furthermore, `.fasta` files do not contain quality information. 
* note all genomes had been trimmed, qc'd and assembled before they were passed on to me for use. 
* using the `genbank_dl.sh` script. This was done on a high throughput computing cluster, where the files are now hosted for the time being. 

**NOTE FASTQC WAS ALREADY PERFORMED PRIOR TO WHEN I RECEIVED THESE ASSEMBLIES**. As was trimming and assembly. 

## Obtaining sequences 
Sequences were obtained from GenBank, and the accessions listed in the directory `./GB_acc_no` using a script (`genbank_dl.sh`) that calls the NCBI e-utils to download each and store in `.fasta` file. This was done for each of the four loci. The scripts `blast_genomes.sh` and `extract_4loci.sh` were used to identify and extract each of the four loci mentioned above, resopectively from 25 genomes from _Amanita muscaria_. These two data sets were then concatenated. 

Note that these genomes are not yet publically available. 

## Alignment 
Each concatenated `.fasta` was aligned separately using two independent alignment softwares. (disucss this more later)

FastTree was used to construct a tree Each alignment was manually inspected, and overly gappy sequences were removed. A list of the removed sequences can be found in the `*_taxa_removed.txt` files in this directory. See my `remove_taxa.sh` script for details on how this was performed. Note that this script calls mafft on any input file it receives to update an alignment from which taxa were removed. 

To prepare taxa for phylogenetic analysis, a small pipeline was written to rename taxa, remove sequences with duplicated headers, and trim the alignment using TrimAl. Details of this can be found in `prep_al.sh`. 

## Constructing MP & Distance Trees  