This project is aimed at generating a multilocus phylogeny for North American species in sect. Amanita (subgen Amanita, genus Amanita Although this is a phylogenetics project, it's ultimate goal is to interrogate the taxonomy of this section, to see how applied names compare with apparent phylogenetic signal.

**Data**: 
* publically available GenBank accessions from several species in section _Amanita_. See `./RawData/GB_acc_no` for the accession numbers and taxa sampled. For each taxon, I have tried to get data across all four loci (ITS1, LSU, $\beta$-Tubulin, and EF1-$\alpha$)
* Four loci data extracted from 25 **unpublished** *Amanita muscaria* genomes from across the globe, including invasive populations
* Four loci data from _A. rubescens_ (subgen. _Amanitina_ sect. _Validae_ ), which I have chosen for it's static relationship to the members of this section.  
* GenBank Accessions were currated manually and downloaded into four `.fasta.` files. These do not have quality scores and were likely generated by sanger sequencing, making automated QC impossible. Furthermore, `.fasta` files do not contain quality information. 
* note all genomes had been trimmed, qc'd and assembled before they were passed on to me for use. 
* using the `genbank_dl.sh` script. This was done on a high throughput computing cluster, where the files are now hosted for the time being. 

**NOTE FASTQC WAS ALREADY PERFORMED PRIOR TO WHEN I RECEIVED THESE ASSEMBLIES**. As was trimming and assembly. 

## Obtaining sequences 
Sequences were obtained from GenBank, and the accessions listed in the directory `./GB_acc_no` using a script (`genbank_dl.sh`) that calls the NCBI e-utils to download each and store in `.fasta` file. This was done for each of the four loci. The scripts `blast_genomes.sh` and `extract_4loci.sh` were used to identify and extract each of the four loci mentioned above, resopectively from 25 genomes from _Amanita muscaria_. These two data sets were then concatenated. 

Note that these genomes are not yet publically available. 

## Alignment 
Each concatenated `.fasta` was aligned separately using two independent alignment softwares. (disucss this more later). 

Following alignment, a custom script, `prep_fa.sh` was used to alter the fasta sequence headers such that they contained the GenBank ID, if applicable, and the full taxon name. 

Each alignment was manually inspected, and the headers for low quality sequences were recorded. Then, FastTree was used to construct a maximum likelihood phylogeny. For each gene tree, members of a given species that did not form a clade with the other members of that species were removed. Indels are common in fungi, especially in the ITS region and intronic sequences. Moreover, the rDNA cistron (which contains both the LSU and ITS sequences) is not single copy. Thus, it seemed appropriate to remove such sequences. The list of the removed sequences can be found in the `*_taxa_removed.txt` files in this directory. See my `remove_taxa.sh` script for details on how this was performed. This script is intended to remove certain sequences from the initial concatenated file and calls mafft once again. 

The `remove_taxa.sh` script also filters out duplicated sequences prior to alignment and trims the alignment with TrimAl. 

## Constructing MP & Distance Trees 
