# Phylogeny of Section _Amanita_ in North America and Europe

This project is aimed at generating a multilocus phylogeny for North American species in sect. Amanita (subgen Amanita, genus Amanita Although this is a phylogenetics project, it's ultimate goal is to interrogate the taxonomy of this section, to see how applied names compare with apparent phylogenetic signal.

## Data  
* publically available GenBank accessions from several species in section _Amanita_. See `./RawData/GB_acc_no` for the accession numbers and taxa sampled. For each taxon, I have tried to get data across all four loci (ITS1, LSU, $\beta$-Tubulin, and EF1-$\alpha$)
* GenBank Accessions were currated manually and downloaded into four `.fasta.` files. These do not have quality scores and were likely generated by sanger sequencing, making automated QC impossible. Furthermore, `.fasta` files do not contain quality information. 
* Though it was originally my intent to include sequences extracted from private genomes belonging to infraspecific taxa identified as _Amanita muscaria_, I decided against this. It did not add much to the phylogeny. Nevertheless, I've decided to leave the steps I took to extract the four loci from the genomes for future reference. 

## Obtaining sequences 
Sequences were obtained from GenBank, and the accessions listed in the directory `./GB_acc_no` using a script (`genbank_dl.sh`) that calls the NCBI e-utils to download each and store in `.fasta` file. This was done for each of the four loci. The scripts `blast_genomes.sh` and `extract_4loci.sh` were used to identify and extract each of the four loci mentioned above, respectively from 25 genomes from _Amanita muscaria_. These two data sets were then concatenated. 

Note that these genomes are not yet publically available. 

## Alignment 
Each concatenated `.fasta` was aligned separately using two independent alignment softwares. (disucss this more later). This alignment was treated as an 

Following alignment, a custom script, `prep_fa.sh` was used to alter the fasta sequence headers such that they contained the GenBank ID, if applicable, and the full taxon name. 

Each alignment was manually inspected, and the headers for low quality sequences were recorded. Then, FastTree, a speedy but somewhat spurious program for inferring maximum likelihood trees in a manner similar to IQTree, was used to construct a preliminary maximum likelihood phylogeny. Note that these trees were prepared only to find highly divergent sequenes. For each gene tree, members of a given species that did not form a clade with the other members of that species were removed. Indels are common in fungi, especially in the ITS region and intronic sequences. Moreover, the rDNA cistron (which contains both the LSU and ITS sequences) is not single copy. Thus, it seemed appropriate to remove such sequences. The list of the removed sequences can be found in the `*_taxa_removed.txt` files in this directory. See my `remove_taxa.sh` script for details on how this was performed. This script is intended to remove certain sequences from the initial concatenated file and calls mafft once again. 

The `remove_taxa.sh` script also filters out duplicated sequences prior to alignment and trims the alignment with TrimAl. 

## Constructing MP & Distance Trees 

Several R libraries were used to prepare