This project is aimed at generating a multilocus phylogeny for North American species in sect. Amanita (subgen Amanita, genus Amanita Although this is a phylogenetics project, it's ultimate goal is to interrogate the taxonomy of this section, to see how applied names compare with apparent phylogenetic signal.

**Data**: 
* publically available GenBank accessions from several species in section _Amanita_. See `./RawData/GB_acc_no` for the accession numbers and taxa sampled. For each taxon, I have tried to get data across all four loci (ITS1, LSU, $\beta$-Tubulin, and EF1-$\alpha$)
* Four loci data extracted from 25 **unpublished** *Amanita muscaria* genomes from across the globe, including invasive populations
* Four loci data from _A. rubescens_ (subgen. _Amanitina_ sect. _Validae_ ), which I have chosen for it's static relationship to the members of this section.  
* GenBank Accessions were currated manually and downloaded into four `.fasta.` files. These do not have quality scores and were likely generated by sanger sequencing, making automated QC impossible. Furthermore, `.fasta` files do not contain quality information. 
* note all genomes had been trimmed, qc'd and assembled before they were passed on to me for use. 
* using the `genbank_dl.sh` script. This was done on a high throughput computing cluster, where the files are now hosted for the time being. 

See `blast_genomes.sh` for the script to blast against each of the aformentioned genomes. The genomes used in this particular project are assemblies from paired end Illumina seq data. 

See the `blastOut_grab.sh` for using the hits to get `.fasta` format data from genomes. 

**NOTE FASTQC WAS ALREADY PERFORMED PRIOR TO WHEN I RECEIVED THESE ASSEMBLIES**. As was trimming and assembly. 

Current state: QC'd, awaiting alignments. 